### PR TITLE
Fixed bug in itembox and added events to datagrid

### PR DESCRIPTION
**Pull request by wachterjohannes**
Main PR: https://github.com/sulu-io/sulu/pull/1347

### CODE REVIEW

**Comment by danrot** [comment]
What was this bug causing?

File: husky_components/datagrid/main.js [hunk 1]
```diff
@@ -495,6 +495,24 @@
             },
 
             /**
+             * deselects an item with a given id
+             * @event husky.datagrid.deselect.item
+             * @param {String|Number} The id of the item to deselect
+             */
+            DESELECT_ITEM = function() {
+                return this.createEventName('deselect.item');
```
Comment (danrot) [inline husky_components/datagrid/main.js]: Don't we have the verb usually last?

File: husky_components/datagrid/main.js [hunk 2]
```diff
@@ -495,6 +495,24 @@
...
+            },
+
+            /**
+             * toggle an item with a given id
+             * @event husky.datagrid.toggle.item
+             * @param {String|Number} The id of the item to toggle
+             */
+            TOGGLE_ITEM = function() {
+                return this.createEventName('toggle.item');
```
Comment (danrot) [inline husky_components/datagrid/main.js]: Same as above, would prefer `item.toggle`, if it is not a big deal.

File: husky_components/datagrid/main.js (hunk 1 above)
Comment (wachterjohannes) [inline husky_components/datagrid/main.js]: hmm all other events in the datagrid has the same style (for my opinion also wrong) - like `husky.datagrid.select.item`. and i think we should`t change all event names ...

File: husky_components/datagrid/main.js (hunk 2 above)
Comment (wachterjohannes) [inline husky_components/datagrid/main.js]: same as above

**Comment by wachterjohannes** [comment]
a request is done with the ids array was empty ... then the loader was not removed and the before items are in the list ... all of this can happen if you deselect all items in overlay!

File: husky_components/datagrid/main.js (hunk 1 above)
Comment (danrot) [inline husky_components/datagrid/main.js]: Ok, then it is just different for all other components. But I agree, changing that would be a too big BC break.

File: husky_extensions/itembox.js [hunk 3]
```diff
@@ -327,7 +327,7 @@ define(function() {
                 // reset items visible when new content is loaded
                 this.viewAll = false;
 
-                if (!!data) {
+                if (!!data && data.length > 0) {
```
Comment (wachterjohannes) [inline husky_extensions/itembox.js]: created new issue with media-selection (is not a array its an object ...)

Comment (danrot) [inline husky_extensions/itembox.js]: I hope the media-selection is working anyway? And can you link the issue here, for reference?

Comment (wachterjohannes) [inline husky_extensions/itembox.js]: yeah media-selection is a object so it has no length (:

**Pull request merged**